docs/developers/how-to/building.rst

```diff
@@ -0,0 +1,81 @@
+==========================
+Build MicroOVN from source
+==========================
+
+Building from source
+--------------------
+
+MicroOVN is distributed as a snap and the only requirements for building it
+are ``Make`` and ``snapcraft``. You can install them with:
+
+.. code-block:: none
+
+   sudo apt install make
+   sudo snap install snapcraft --classic
+
+Snapcraft requires ``LXD`` to build snaps. So if your system does not have LXD
+installed and initiated, you can check out either `LXD getting started
+guides`_ or go with following default setup:
+
+.. code-block:: none
+
+   sudo snap install lxd
+   lxd init --auto
+
+To build MicroOVN, go into the repository's root directory and run:
+
+.. code-block:: none
+
+   make
+
+This will produce the ``microovn.snap`` file that can be then used to install
+MicroOVN on your system.
+
+Installing from local file
+--------------------------
+
+Following up on the previous section, you can install the ``microovn.snap``
+file with:
+
+.. code-block:: none
+
+   snap install --dangerous ./microovn.snap
+
+.. note::
+
+   If you are building latest MicroOVN from the ``main`` branch, it's possible
```

Comment:
i'm wondering whether we can pre-empt this in the requirements section? is there a build scenario that we should be recommending that will enable us to give the user everything they need from the start?

Reply:
This is situational. Someone might be building from stable branch and they wouldn't encounter this requirement. Also, as we get closer to 24.04 release, the `core24` will become stable and this step wont be needed until we get closer to the next release and we'll switch our `main` branch to using the next pre-release core.
I think it's OK to leave it here as a note since it's a possibility.